pkg/controller/data_download_controller.go

```diff
@@ -382,15 +398,12 @@ func (r *DataDownloadReconciler) SetupWithManager(mgr ctrl.Manager) error {
 func (r *DataDownloadReconciler) findSnapshotRestoreForPod(podObj client.Object) []reconcile.Request {
 	pod := podObj.(*v1.Pod)
 
-	dd := &velerov2alpha1api.DataDownload{}
-	err := r.client.Get(context.Background(), types.NamespacedName{
-		Namespace: pod.Namespace,
-		Name:      pod.Labels[velerov1api.DataDownloadLabel],
-	}, dd)
-
+	dd, err := findDataDownloadByPod(r.client, *pod)
 	if err != nil {
 		r.logger.WithField("Restore pod", pod.Name).WithError(err).Error("unable to get DataDownload")
 		return []reconcile.Request{}
+	} else if dd == nil {
+		return []reconcile.Request{}
```

Comment:
Add an error log here as well? Or assert dd must not be nil when err is nil?

Reply:
it's possible  for `dd == nil`, which is different from `err == nil`
 
 I'll add one log here

Reply:
Added